ts/src/base/Exchange.ts

```diff
@@ -864,17 +882,65 @@ export default class Exchange {
                 headers = this.extend (this.userAgent, headers)
             }
         }
-        if (typeof this.proxy === 'function') {
-            url = (this as any).proxy (url)
-            if (isNode) {
-                headers = this.extend ({ 'Origin': this.origin }, headers)
+
+        // ################## PROXY ##################
+        // old approach, kept for backward-compatibility
+        if (this.proxy !== undefined) {
+            if (typeof this.proxy === 'function') {
+                url = (this as any).proxy (url)
+                if (isNode) {
+                    headers = this.extend ({ 'Origin': this.origin }, headers)
+                }
+            } else if (typeof this.proxy === 'string') {
+                if (this.proxy.length && isNode) {
+                    headers = this.extend ({ 'Origin': this.origin }, headers)
+                }
+                url = this.proxy + url
             }
-        } else if (typeof this.proxy === 'string') {
-            if (this.proxy.length && isNode) {
-                headers = this.extend ({ 'Origin': this.origin }, headers)
+        }
+        // new code
+        else {
+            const [ proxyUrl, proxyUrlCallback, proxyHttp, proxyHttps, proxySocks, proxyAgentCallback ] = this.checkProxySettings ();
+            if (proxyUrl !== undefined) {
+                // in node we need to set header to *
+                if (isNode) {
+                    headers = this.extend ({ 'Origin': this.origin }, headers)
+                }
+                url = proxyUrl + url
+            } else if (proxyUrlCallback !== undefined) {
+                // in node we need to set header to *
+                if (isNode) {
+                    headers = this.extend ({ 'Origin': this.origin }, headers)
+                }
+                url = proxyUrlCallback (url, method, headers, body);
+            } else if (proxyHttp !== undefined) {
+                const module = await import (/* webpackIgnore: true */ '../static_dependencies/proxies/http-proxy-agent/index.js')
```

Comment:
We're doing here dynamic imports, we have to make sure the CJS bundles work with them

Reply:
I've tested and it worked on my side